### PR TITLE
refactor(avatar): Refactor avatar/logo upload

### LINKED ISSUE #380: Refactor avatar/logo upload code
Now the originally selected picture is uploaded to GAE, then that image is shown such that the user can crop it. Refactor this such that the initial upload step is skipped. We already did this for news items, there we use a html 5 canvas.

Do this for the solution avatar and main branding header image

### PR DESCRIPTION
**Pull request by lucasvanhalst**
No longer store the images on the server while cropping the image.

Migration:

```python
def remove_temp_blobs():
    from google.appengine.ext import ndb
    ndb.delete_multi(ndb.Query(kind='TempBlob').fetch(keys_only=True))
```

Closes #380 

Should be merged together with rogerthat-platform/rogerthat-backend/pull/232

### CODE REVIEW

File: src/solutions/common/bizz/settings.py [hunk 1]
```diff
@@ -179,8 +157,7 @@ def trans():
 
         return settings
 
-    xg_on = db.create_transaction_options(xg=True)
-    common_settings = db.run_in_transaction_options(xg_on, trans)
+    common_settings = trans()
```
Comment (RubenMattan) [inline src/solutions/common/bizz/settings.py]: transaction

File: src/solutions/common/bizz/settings.py [hunk 2]
```diff
@@ -206,6 +184,5 @@ def trans():
         put_and_invalidate_cache(sln_main_branding, common_settings)
         return common_settings
 
-    xg_on = db.create_transaction_options(xg=True)
-    common_settings = db.run_in_transaction_options(xg_on, trans)
+    common_settings = trans()
```
Comment (RubenMattan) [inline src/solutions/common/bizz/settings.py]: transaction

**Comment by RubenMattan** [comment]
Failed to publish...

```python

ERROR    2017-07-17 14:16:18,904 add_1_monkey_patches.py:111] Failure in common_provision

Traceback (most recent call last):
  File "/Users/ruben/mobicage/git/oca-backend/build/solutions/common/bizz/__init__.py", line 600, in common_provision
    needs_reload = bizz.provision(service_user, friends)
  File "/Users/ruben/mobicage/git/oca-backend/build/mcfw/rpc.py", line 164, in typechecked_return
    result = f(*args, **kwargs)
  File "/Users/ruben/mobicage/git/oca-backend/build/mcfw/rpc.py", line 142, in typechecked_f
    return f(**kwargs)
  File "/Users/ruben/mobicage/git/oca-backend/build/solutions/flex/bizz.py", line 113, in provision
    put_avatar_if_needed(service_user)
  File "/Users/ruben/mobicage/git/oca-backend/build/mcfw/rpc.py", line 164, in typechecked_return
    result = f(*args, **kwargs)
  File "/Users/ruben/mobicage/git/oca-backend/build/mcfw/rpc.py", line 142, in typechecked_f
    return f(**kwargs)
  File "/Users/ruben/mobicage/git/oca-backend/build/solutions/common/bizz/provisioning.py", line 365, in put_avatar_if_needed
    system.put_avatar(base64.b64encode(sln_avatar.picture))
  File "/Users/ruben/mobicage/git/oca-backend/build/rogerthat/rpc/service.py", line 346, in wrapped
    result = f(*args, **kwargs)
  File "/Users/ruben/mobicage/git/oca-backend/build/mcfw/rpc.py", line 164, in typechecked_return
    result = f(*args, **kwargs)
  File "/Users/ruben/mobicage/git/oca-backend/build/mcfw/rpc.py", line 142, in typechecked_f
    return f(**kwargs)
  File "/Users/ruben/mobicage/git/oca-backend/build/rogerthat/service/api/system.py", line 446, in put_avatar
    update_service_avatar(service_user, image1)
  File "/Users/ruben/mobicage/git/oca-backend/build/mcfw/rpc.py", line 164, in typechecked_return
    result = f(*args, **kwargs)
  File "/Users/ruben/mobicage/git/oca-backend/build/mcfw/rpc.py", line 142, in typechecked_f
    return f(**kwargs)
  File "/Users/ruben/mobicage/git/oca-backend/build/rogerthat/bizz/profile.py", line 796, in update_service_avatar
    return trans()
  File "/Users/ruben/google-cloud-sdk/platform/google_appengine/google/appengine/api/datastore.py", line 2749, in inner_wrapper
    return RunInTransactionOptions(options, func, *args, **kwds)
  File "/Users/ruben/google-cloud-sdk/platform/google_appengine/google/appengine/api/datastore.py", line 2647, in RunInTransactionOptions
    ok, result = _DoOneTry(function, args, kwargs)
  File "/Users/ruben/google-cloud-sdk/platform/google_appengine/google/appengine/api/datastore.py", line 2667, in _DoOneTry
    result = function(*args, **kwargs)
  File "/Users/ruben/mobicage/git/oca-backend/build/rogerthat/bizz/profile.py", line 791, in trans
    service_profile.put()
  File "/Users/ruben/mobicage/git/oca-backend/build/mcfw/cache.py", line 57, in put
    self._trigger_invalidate_cache()
  File "/Users/ruben/mobicage/git/oca-backend/build/mcfw/cache.py", line 51, in _trigger_invalidate_cache
    self.on_trans_committed(invalidate_cache)
  File "/Users/ruben/mobicage/git/oca-backend/build/add_2_zip_imports.py", line 61, in <lambda>
    CachedModelMixIn.on_trans_committed = lambda self, f, *args, **kwargs: on_trans_committed(f, *args, **kwargs)
  File "/Users/ruben/mobicage/git/oca-backend/build/rogerthat/utils/transactions.py", line 83, in on_trans_committed
    post_transaction_actions.append(True, func, *args, **kwargs)
  File "/Users/ruben/mobicage/git/oca-backend/build/rogerthat/utils/transactions.py", line 45, in append
    transaction_guid = self.get_current_transaction_guid()
  File "/Users/ruben/mobicage/git/oca-backend/build/rogerthat/utils/transactions.py", line 38, in get_current_transaction_guid
    return self.transaction_guids[-1]
IndexError: list index out of range
```

File: src/solutions/common/bizz/settings.py (hunk 1 above)
Comment (lucasvanhalst) [inline src/solutions/common/bizz/settings.py]: See @db.transactional

**Comment by bart-at-mobicage** [comment]
probably accidently closed while closing everything of milestone bug-hunt-1

**Pull request merged**